pints/_mcmc/__init__.py

```diff
@@ -293,13 +293,60 @@ class MCMCController(object):
     method : class
         The class of :class:`MCMCSampler` to use. If no method is specified,
         :class:`HaarioBardenetACMC` is used.
+    transform : pints.Transform
+        A :class:`pints.Transform` that transform the model parameter space to
+        search space. If no transform is specified, search will be performed
+        in the model space.
     """
 
-    def __init__(self, log_pdf, chains, x0, sigma0=None, method=None):
+    def __init__(
+            self, log_pdf, chains, x0, sigma0=None, method=None,
+            transform=None):
 
-        # Store function
+        # Check function
         if not isinstance(log_pdf, pints.LogPDF):
             raise ValueError('Given function must extend pints.LogPDF')
+
+        # Transform everything
+        # From this point onward the MCMC sampler will see only the
+        # transformed search space and will know nothing about the model
+        # parameter space.
+        if transform:
+            log_pdf = transform.apply_log_pdf(log_pdf)
+            x0 = [transform.to_search(x) for x in x0]
+            #
+            # Following Eq. 5 in
+            # http://citeseerx.ist.psu.edu/viewdoc/summary?doi=10.1.1.47.9023
+            #
+            # For transformation x = g(p) and Jacobian J(x) = d/dx g^{-1}(x),
+            # then the covariance matrices C(.) for x and p follow
+            #
+            # C(x) = (d/dp g(E(p))) C(p) (d/dp g(E(p)))^T
+            #        = J^{-1}(E(x)) C(p) J^{-1}(E(x))^T
+            #
+            # Using the property that J^{-1} = dg/dp, from the inverse function
+            # theorem, i.e. the matrix inverse of the Jacobian matrix of an
+            # invertible function is the Jacobian matrix of the inverse
+            # function.
+            #
+            if sigma0 is not None:
+                # Transform sigma0 if provided
+                sigma0 = np.asarray(sigma0)
+                n_parameters = log_pdf.n_parameters()
+                # Make sure sigma0 is a (covariance) matrix
+                if np.product(sigma0.shape) == n_parameters:
+                    # Convert from 1d array
+                    sigma0 = sigma0.reshape((n_parameters,))
+                    sigma0 = np.diag(sigma0)
+                else:
+                    # Check if 2d matrix of correct size
+                    sigma0 = sigma0.reshape((n_parameters, n_parameters))
+                # Transform sigma0
```

Comment:
This is the only part of the "if sigma" block that's part of the transformation handling

Reply:
What do you mean here?